src/components/PriceCheckForm.js

```diff
@@ -1,33 +1,70 @@
 import React, { Component } from 'react'
+import axios from 'axios';
+import { categories } from '../utils/Constants.js';
+
 
 class PriceCheckForm extends Component {
     constructor(props){
         super(props)
         this.state = {
-            postcode: ""
+            postcode: "",
+            selectedCategory: "",
+            searchString: ""
         }
 
         this.handleSubmit = this.handleSubmit.bind(this);
-        this.handleChange = this.handleChange.bind(this);
-       
+        this.handlePostcodeChange = this.handlePostcodeChange.bind(this);
+        this.handleCategoryChange = this.handleCategoryChange.bind(this);
+        this.handleSearchStringChange = this.handleSearchStringChange.bind(this);
+        this.fetchSuppliers = this.fetchSuppliers.bind(this)
+    }
+
+    fetchSuppliers = (postcode) => {
+        return axios.get(`https://www.foodbomb.com.au/supplier-service/postcode/${postcode}/suppliers`)
+          .then(resp => resp.data)
     }
 
     handleSubmit(e) {
-        console.log(`Submitted the postcode ${this.state.postcode} for query`)
+        const supplierResponse = this.fetchSuppliers(this.state.postcode)
         e.preventDefault()
     }
 
-    handleChange(e) {
-        this.setState({postcode: e.target.value})
+
+    handlePostcodeChange(e) {
+        this.setState({postcode: e.target.value})   
+    }
+
+    handleCategoryChange(e) {
+        this.setState({selectedCategory: categories[e.target.id-1].value})
+        console.log(`${this.state.selectedCategory}`)
+    }
+
+    handleSearchStringChange(e) {
+        this.setState({searchString: e.target.value}) 
     }
 
+
     render() {
+        const categoryRadioSelectors = categories.map( category => 
+            <div key={category.catId} onChange={this.handleCategoryChange}>
+                <input type="radio" id={category.catId} name="category" value={category.value}/>
+                <label for={category.catId}>{category.displayName}</label>
+            </div>
```

Comment:
Nice map
We could online this directly into the render function, it’s not that massive that we need to store it in a variable